docs/design/queue-resource-reservation-design-v2.md

```diff
@@ -0,0 +1,35 @@
+# Volcano Resource Reservation For Queue(v2)
+
+@[Thor-wl](https://github.com/Thor-wl); Feb 28th, 2021
+
+Inspired by design optimization of `target job resource reservation`, our goal is to ensure the SLA of jobs in target
+queues. So just allocate resource for jobs by order SLA.
+
+## Design
+Queues with annotation `maxWaitingTime` indicate that this queue should ensure SLA. Jobs in the queue will be scheduled
+first if its SLA 
+
+## Implementation
+### API
+```
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: q1
+  annotations:
+    # maxWaitingTime indicates the maximal pending time for jobs in the queue. 
+    # If not scheduled after waiting for maxWaitTime, jobs will be scheduled first.
+    maxWaitingTime: 30m
```

Comment:
if we add `maxWaitingTime` in queue annotation, then any user can set SLA on queue, which should only be modified by admin in my opinion. So would it be better to set `maxWaitingTime` in some other field like `status`?

Reply:
> if we add `maxWaitingTime` in queue annotation, then any user can set SLA on queue, which should only be modified by admin in my opinion. So would it be better to set `maxWaitingTime` in some other field like `status`?

Yes. Do you have any other good ideas?